pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -1487,6 +1498,33 @@ static void updateColumnNames(String rawTableName, PinotQuery pinotQuery, boolea
     }
   }
 
+  private static void expandStarExpressionsToActualColumns(PinotQuery pinotQuery, Map<String, String> columnNameMap,
+      Expression selectStarExpr) {
+    List<Expression> originalSelections = pinotQuery.getSelectList();
+    //expand '*'
+    List<Expression> expandedSelections = new ArrayList<>();
+    for (String tableCol : columnNameMap.values()) {
+      Expression newSelection = RequestUtils.createIdentifierExpression(tableCol);
+      //we exclude default virtual columns
+      if (tableCol.charAt(0) != '$') {
+        expandedSelections.add(newSelection);
+      }
+    }
+    //sort naturally
+    expandedSelections.sort(null);
+    ListIterator<Expression> li = originalSelections.listIterator();
```

Comment:
Modifying the existing list can be expensive (keep shifting the values). We can create a new list to replace the original one

Reply:
Done, thanks!